src/main/java/io/openliberty/website/TLSFilter.java

```diff
@@ -125,12 +136,30 @@ public void doFilter(ServletRequest req, ServletResponse resp, FilterChain chain
                 }
                 if (doGzip) {
                     response.setHeader("Content-Type", "text/html");
-                    response.setHeader("Content-Encoding", "gzip");
                     doFilter = false;
                     try {
-                        req.getRequestDispatcher(uri.concat(".gz")).include(req, response);
+                        if (queryString != null) {
+                            uriQueryString = uri +"?"+ queryString;
+                            docsPage = Jsoup.connect("https://openliberty.io"+uriQueryString).get();
+                            String ifError = docsPage.select("head > title").first().text();
+                            if(ifError.contains("404")){
+                                response.setStatus(HttpServletResponse.SC_NOT_FOUND);
+                                response.sendRedirect("/404.html");
+                            }
+                            else {
+                                String updatedCanonicalUrl = uriQueryString.replace(uriQueryString.split("\\/")[2], "latest");
+                                Element result = docsPage.select("link[rel=\"canonical\"]").first();
+                                result.attr("href", "https://openliberty.io"+updatedCanonicalUrl);
+                                htmlResponse = response.getWriter();
+                                htmlResponse.println(docsPage.html());
```

Comment:
Do we need to close or flush any kind of stream when using `PrintWriter`? We want to confirm that we do not need to call anything to "close" the writer so we do not have a memory leak that can crash the server.

Reply:
added flush and close